src/screens/Forum/ForumHomeScreen.js

```diff
@@ -35,6 +35,12 @@ export default class ForumHomeScreen extends React.Component {
           const time = date ? date.toTimeString() : null;
           const { currentUserID } = this.state;
 
+          // Set state to be passed in navigateToPostScreen
+          if (post.userID) {
+            this.setState({ userID: post.userID });
+          }
+          this.setState({ postID: post.id });
+
```

Comment:
This doesn't work as intended; the state here refers to `ForumHomeScreen`'s state, so `userID` and `postID` just keep getting updated over and over in this `map()`. That means when you finally read the state in `navigateToPostScreen()`, you will always read the same values (the `userID` and `postID` of the last post in this array that you're mapping over).

Instead let's remove these lines, make `navigateToPostScreen` take userID and postID as parameters, and then when we actually call this function in `ForumPost`, we can give it the correct arguments there.